Sources/App/Migrations/054/CreateWeightedKeywords.swift

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2021 Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Fluent
+import SQLKit
+
+struct CreateWeightedKeywords: Migration {
+    func prepare(on database: Database) -> EventLoopFuture<Void> {
+        guard let db = database as? SQLDatabase else {
+            fatalError("Database must be an SQLDatabase ('as? SQLDatabase' must succeed)")
+        }
+
+        return db.raw(
+            """
+            -- v0
+            CREATE MATERIALIZED VIEW weighted_keywords AS
+            SELECT COUNT(*) AS count, UNNEST(keywords) AS keyword
+            FROM repositories
+            GROUP BY keyword
+            """).run()
+    }
+
+    func revert(on database: Database) -> EventLoopFuture<Void> {
+        guard let db = database as? SQLDatabase else {
+            fatalError("Database must be an SQLDatabase ('as? SQLDatabase' must succeed)")
+        }
+
+        return db.raw("DROP MATERIALIZED VIEW weighted_keywords").run()
+    }
+}
```

Comment:
I commented on the commit itself a moment ago on my iPad but now I can't see where that actually went. The problem wasn't that the migration was async - it should actually be, for future-proofing. The problem was that the initial implementation had a `db.raw(dropSQL)` statement in the `prepare` step. Since this is the _first_ migration that creates the view that will throw the error. I'm guessing you'd copied this from a search view where it's updating an existing search view - and that's why that was necessary in that case.

Reply:
I see. I will make sure the migration is like this.

Reply:
And I have been squashing and force pushing on this branch. So you are not crazy. 😇